frontend/src/metabase/query_builder/components/QueryModals.tsx

```diff
@@ -277,17 +277,19 @@ class QueryModals extends Component<QueryModalsProps> {
             <EntityCopyModal
               entityType="questions"
               entityObject={{
-                ...question.card(),
+                ...this.props.card,
                 collection_id: question.canWrite()
                   ? question.collectionId()
                   : initialCollectionId,
               }}
               copy={async formValues => {
                 const object = await this.props.onCreate(
-                  question
-                    .setDisplayName(formValues.name)
-                    .setCollectionId(formValues.collection_id)
-                    .setDescription(formValues.description || null),
+                  new Question({
```

Comment:
The root cause is that `getQuestion` selector https://github.com/metabase/metabase/blob/3400a07688ecfcded85eea65570335f04de4aca0/frontend/src/metabase/query_builder/selectors.js#L318 automatically wraps the question if it's a model with `composeDataset` in the `view` mode ( = simple/chill mode). This works for many cases, but doesn't work for duplicating a model.

I think there are 2 ways how this can be solved:
1. Add another selector that returns a `Question` instance before the `composeDataset` call, then make the original `getQuestion` selector use the new one. Adjust the copy modal to use the correct selector.
2. "Unwrap" the source query using something like `question.isDataset() ? question.query().sourceQuery().question() : question`. This feels wrong to me, and I'd rather go with #1.

Wdyt?

Reply:
I think the selector solution does seem to be the best - I agree that the second option feels a little hacky at this stage. I'll have a go at it